src/test/blockchain/blockchain_behavior_tests.cpp

```diff
@@ -39,4 +49,92 @@ BOOST_AUTO_TEST_CASE(creation_test) {
   }
 }
 
+BOOST_AUTO_TEST_CASE(GetTransactionWeight_test) {
+  mocks::ArgsManagerMock args{};
+  const auto b = blockchain::Behavior::New(&args);
+
+  {
+    // Check weight of empty transaction
+    CTransaction tx;
+
+    const std::size_t expected = GetTransactionWeight(tx);
+    const std::size_t weight = b->GetTransactionWeight(tx);
+
+    BOOST_CHECK_EQUAL(40, weight);
+    BOOST_CHECK_EQUAL(expected, weight);
+  }
+
+  txtools::TxTool key_tool;
+  {
+    // Check weight of standard transaction
+    const CTransaction tx = key_tool.CreateTransaction();
+
+    const std::size_t expected = GetTransactionWeight(tx);
+    const std::size_t weight = b->GetTransactionWeight(tx);
+
+    BOOST_CHECK_EQUAL(437, weight);
```

Comment:
Failed several times with values ranging from 435 to 437

Reply:
I will adapt